### ISSUE TITLE
WebContent crash causes segfault in main process

### ISSUE THREAD

**Issue by mjakeman** [mjakeman/libweb-gtk#1]
When the WebContent process crashes, it also takes down the main process

This is probably due to a dangling pointer somewhere

**Comment by mjakeman** [comment]
This is also related to why the process crashes on shutdown:

```
frame #0: 0x0000010276d30c00
    frame #1: 0x0000000101cd4b08 liblagom-webview.0.dylib`::clear() [inlined] unref at RefCounted.h:65:13 [opt]
    frame #2: 0x0000000101cd4ae8 liblagom-webview.0.dylib`::clear() [inlined] unref_if_not_null<Core::Notifier> at NonnullRefPtr.h:32:14 [opt]
    frame #3: 0x0000000101cd4ae4 liblagom-webview.0.dylib`::clear() [inlined] clear at RefPtr.h:223:9 [opt]
    frame #4: 0x0000000101cd4adc liblagom-webview.0.dylib`::clear() [inlined] ~RefPtr at RefPtr.h:103:9 [opt]
    frame #5: 0x0000000101cd4adc liblagom-webview.0.dylib`::clear() [inlined] ~RefPtr at RefPtr.h:102:5 [opt]
    frame #6: 0x0000000101cd4adc liblagom-webview.0.dylib`::clear() [inlined] ~PosixSocketHelper at Socket.h:106:7 [opt]
    frame #7: 0x0000000101cd4adc liblagom-webview.0.dylib`::clear() [inlined] ~PosixSocketHelper at Socket.h:106:7 [opt]
    frame #8: 0x0000000101cd4adc liblagom-webview.0.dylib`::clear() [inlined] ~LocalSocket at Socket.h:342:39 [opt]
    frame #9: 0x0000000101cd4ad4 liblagom-webview.0.dylib`::clear() [inlined] ~LocalSocket at Socket.h:342:28 [opt]
    frame #10: 0x0000000101cd4ad4 liblagom-webview.0.dylib`::clear() [inlined] operator() at DefaultDelete.h:17:9 [opt]
    frame #11: 0x0000000101cd4ad0 liblagom-webview.0.dylib`::clear() at OwnPtr.h:110:9 [opt]
    frame #12: 0x0000000101cd4aa0 liblagom-webview.0.dylib`::~ConnectionBase() [inlined] ~OwnPtr at OwnPtr.h:45:9 [opt]
    frame #13: 0x0000000101cd4a9c liblagom-webview.0.dylib`::~ConnectionBase() [inlined] ~OwnPtr at OwnPtr.h:44:5 [opt]
    frame #14: 0x0000000101cd4a9c liblagom-webview.0.dylib`::~ConnectionBase() at Connection.h:39:48 [opt]
    frame #15: 0x0000000101cd22c4 liblagom-webview.0.dylib`::~WebContentClient() [inlined] ~Connection at Connection.h:86:7 [opt]
    frame #16: 0x0000000101cd22bc liblagom-webview.0.dylib`::~WebContentClient() [inlined] ~ConnectionToServer at ConnectionToServer.h:29:7 [opt]
    frame #17: 0x0000000101cd2288 liblagom-webview.0.dylib`::~WebContentClient() [inlined] ~WebContentClient at WebContentClient.h:19:7 [opt]
    frame #18: 0x0000000101cd227c liblagom-webview.0.dylib`::~WebContentClient() [inlined] ~WebContentClient at WebContentClient.h:19:7 [opt]
    frame #19: 0x0000000101cd227c liblagom-webview.0.dylib`::~WebContentClient() at WebContentClient.h:19:7 [opt]
    frame #20: 0x000000010002c898 ladybird`::operator=() [inlined] unref at RefCounted.h:65:13 [opt]
    frame #21: 0x000000010002c878 ladybird`::operator=() [inlined] unref_if_not_null<WebView::WebContentClient> at NonnullRefPtr.h:32:14 [opt]
    frame #22: 0x000000010002c874 ladybird`::operator=() [inlined] clear at RefPtr.h:223:9 [opt]
    frame #23: 0x000000010002c874 ladybird`::operator=() [inlined] ~RefPtr at RefPtr.h:103:9 [opt]
    frame #24: 0x000000010002c874 ladybird`::operator=() [inlined] ~RefPtr at RefPtr.h:102:5 [opt]
    frame #25: 0x000000010002c874 ladybird`::operator=() [inlined] operator= at RefPtr.h:131:5 [opt]
    frame #26: 0x000000010002c864 ladybird`::operator=() at ViewImplementation.h:190:12 [opt]
    frame #27: 0x00000001000292ec ladybird`::create_client() at WebContentView.cpp:543:20 [opt]
    frame #28: 0x0000000101cc9d18 liblagom-webview.0.dylib`::handle_web_content_process_crash() at ViewImplementation.cpp:333:5 [opt]
    frame #29: 0x0000000100032d64 ladybird`::call() [inlined] operator() at WebContentView.cpp:552:13 [opt]
    frame #30: 0x0000000100032d3c ladybird`::call() at Function.h:173:20 [opt]
    frame #31: 0x0000000102781734 liblagom-core.0.dylib`::process() [inlined] operator() at Function.h:111:25 [opt]
    frame #32: 0x000000010278171c liblagom-core.0.dylib`::process() at ThreadEventQueue.cpp:108:13 [opt]
    frame #33: 0x000000010112a574 libglib-2.0.0.dylib`g_timeout_dispatch + 36
    frame #34: 0x000000010112dd44 libglib-2.0.0.dylib`g_main_context_dispatch + 256
    frame #35: 0x000000010112e064 libglib-2.0.0.dylib`g_main_context_iterate + 436
    frame #36: 0x000000010112e384 libglib-2.0.0.dylib`g_main_loop_run + 268
    frame #37: 0x000000010002443c ladybird`::exec() at EventLoopImplementationGLib.cpp:47:5 [opt]
    frame #38: 0x00000001027628cc liblagom-core.0.dylib`::exec() at EventLoop.cpp:75:20 [opt]
    frame #39: 0x0000000100034d2c ladybird`::serenity_main() at main.cpp:168:23 [opt]
    frame #40: 0x0000000100036664 ladybird`main at Main.cpp:39:19 [opt]
    frame #41: 0x0000000186fb3f28 dyld`start + 2236
```

**Comment by mjakeman** [comment]
Fixed by 9bf8acc
